frontend/src/metabase/search/components/SearchResult.jsx

```diff
@@ -179,34 +253,83 @@ function DashboardResult({ dashboard, options }) {
 function QuestionResult({ question, options }) {
   return (
     <ResultLink to={question.getUrl()} compact={options.compact}>
-      <Flex align="center">
-        <ItemIcon item={question} />
+      <Flex align="start">
+        <ItemIcon item={question} type="question" />
         <Box>
           <Title>{question.name}</Title>
-          {formatCollection(question.getCollection())}
+          <Text>{jt`Saved question in ${formatCollection(
+            question.getCollection(),
+          )}`}</Text>
+          {question.description && (
+            <Description>{question.description}</Description>
+          )}
           <Score scores={question.scores} />
         </Box>
-        {question.description && (
-          <Box ml="auto">
-            <Tooltip tooltip={question.description}>
-              <Icon name="info" />
-            </Tooltip>
-          </Box>
-        )}
       </Flex>
       {formatContext(question.context, options.compact)}
     </ResultLink>
   );
 }
 
+function SegmentResult({ segment, options }) {
+  return (
+    <ResultLink to={segment.getUrl()} compact={options.compact}>
+      <Flex align="start">
+        <ItemIcon item={segment} />
+        <Box>
+          <Title>{segment.name}</Title>
+          <Text>{jt`Segment of ${(
+            <Link
+              to={Urls.tableRowsQuery(segment.database_id, segment.table_id)}
+            >
+              <Table.Name id={segment.table_id} />
+            </Link>
+          )}`}</Text>
+          {segment.description && (
+            <Description>{segment.description}</Description>
+          )}
+          <Score scores={segment.scores} />
+        </Box>
+      </Flex>
+    </ResultLink>
+  );
+}
+
+function MetricResult({ metric, options }) {
+  return (
+    <ResultLink to={metric.getUrl()} compact={options.compact}>
+      <Flex align="start">
+        <ItemIcon item={metric} />
+        <Box>
+          <Title>{metric.name}</Title>
+          <Text>{jt`Metric for ${(
+            <Link to={Urls.tableRowsQuery(metric.database_id, metric.table_id)}>
+              <Table.Name id={metric.table_id} />
```

Comment:
This unfortunately isn't returning the "displayName" and is instead returning the original name (e.g. ORDERS instead of Orders).